ui/components/component-library/header-base/header-base.d.ts

```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import { BoxProps } from '../../ui/box';
+
+export interface HeaderBaseProps extends BoxProps {
+  children?: React.ReactNode;
+  childrenWrapperProps?: BoxProps;
+  startAccessory?: React.ReactNode;
+  startAccessoryWrapperProps?: BoxProps;
+  endAccessory?: React.ReactNode;
+  endAccessoryWrapperProps?: BoxProps;
+  className?: string;
+}
```

Comment:
I think we probably want to follow mobiles lead on file conventions and keep our constants like enums and such in a `.constants.ts` and our types in `.types.ts` file. The `.d.ts` file was just for the box component because it's still javascript. I need to update the badge-wrapper to include both.